### PR TITLE
add breadcrumbs component

### PR DESCRIPTION
**Pull request by amalasyrafs**
- added breadcrumbs component
- added stories for breadcrumbs

Preview in storybook:
image

TODO:
- [ ]  write dev docs for breadcrumbs in fumadocs (will proceed once component is approved)

### CODE REVIEW

File: packages/react/src/components/breadcrumbs.tsx
Comment (irfancoder) [inline packages/react/src/components/breadcrumbs.tsx:1]: 1. Rename variant option: ~~alternative~~ to `fill`
2. Rename component: Breadcrumbs* to Breadcrumb* -- eg. `BreadcrumbsItem` => `BreadcrumbItem` and so on.

File: packages/react/src/components/breadcrumbs.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  className,
+  ...props
+}: React.ComponentProps<"li">) => (
+  <li
+    role="presentation"
+    aria-hidden="true"
+    className={clx("[&>svg]:stroke-bg-black-400 [&>svg]:size-6", className)}
+    {...props}
+  >
+    {children ?? <ChevronRight />}
```
Comment (irfancoder) [inline packages/react/src/components/breadcrumbs.tsx:96]: Use `||` instead. Wider scope for checking falsy value when rendering component

https://medium.com/nerd-for-tech/the-difference-between-and-operators-in-javascript-nodejs-3696b0ce02ff

**Review by irfancoder — CHANGES_REQUESTED** [review]
Good and very clean work! Good to merge after feedbacks are addressed

**Comment by amalasyrafs** [comment]
feedback addressed in the new commit

**Review by irfancoder — APPROVED** [review]

**Pull request merged**